### ISSUE TITLE
MySQL - Repository Database

### ISSUE THREAD

**Issue by Andres8427** [zhicwu/docker-biserver-ce#44]
Hello, I thank you and congratulate you for your great contribution with this repository.

I am trying to use MySQL as a repository (not to use HSQLDB by default in Pentaho), however I do not get it to work, modifying the information in "secret.env" and modifying the file "docker-compose.yml" returns me as error:

====================
"Pentaho Initialization Exception

The following errors were detected
One or more system listeners failed. These are set in the systemListeners.xml.
    org.pentaho.platform.api.engine.PentahoSystemException: PentahoSystem.ERROR_0014 - Error while trying to execute startup sequence for org.pentaho.platform.scheduler2.quartz.EmbeddedQuartzSystemListener "
====================

I see that "docker-entrypoint.sh" contains the replacements that should be done to convert MySQL as a repository, but I can not make it work.

Can you please tell me a general step by step so that I can validate that this is wrong?

Thank you.

**Comment by zhicwu** [comment]
Sorry for late reply @Andres8427. You may pull latest 7.1 image and follow instructions at [here](https://github.com/zhicwu/docker-biserver-ce/tree/7.1/examples/mysql).

**Comment by Andres8427** [comment]
Hello and thanks for replying

I followed the indications: I pulled again from repository 7.1, and being in the folder "/examples/mysql/" modify with my connection data the `.env` file and being from the same folder run "docker-compose up"

I see that I apparently do the replacement of the variables in the file repository.xml, context.xml, and others. But not yet initiation, now shows as error:

"

> Pentaho Initialization Exception
> 
> 
> The following errors were detected
> One or more system listeners failed. These are set in the systemListeners.xml.
>    org.pentaho.platform.api.engine.PentahoSystemException: PentahoSystem.ERROR_0014 - Error while trying to execute startup sequence for org.pentaho.platform.engine.services.connection.datasource.dbcp.DynamicallyPooledDatasourceSystemListener
> 
> Please see the server console for more details on each error detected.

 "

Seeing the "logs" of pentaho I find between all the lines a pair that catches my attention

"

> Caused by: org.apache.jackrabbit.core.fs.FileSystemException: failed to initialize file system
>         at org.apache.jackrabbit.core.fs.db.DatabaseFileSystem.init (DatabaseFileSystem.java:207)
>         at org.apache.jackrabbit.core.config.RepositoryConfigurationParser $ 6.getFileSystem (RepositoryConfigurationParser.java:1165)
>         ... 120 more
> Caused by: org.apache.commons.dbcp.SQLNestedException: Can not create PoolableConnectionFactory (Communications link failure
> 
> The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.)

 "

and lower

"

> Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure
> 
> The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.

 "

But I'm sure the connection data is correct.

I'm doing the tests on an Amazon AWS m4.xlarge instance with MySQL installed on the same EC2 (No RDS) with Ubuntu 16.04.

Did something similar happen to you?

**Comment by zhicwu** [comment]
No, it works for me. Which version of MySQL you’re using? Let me add MySQL service in docker-compose file to make this less painful.

**Comment by Andres8427** [comment]
My version of MySQL is 5.7.20

**Comment by zhicwu** [comment]
Please try the simplified example at [here](https://github.com/zhicwu/docker-biserver-ce/tree/7.1/examples/mysql).

**Comment by hieplq** [comment]
i do test pentaho 8.0.2 (the latest public release) and get same same error
i test with default and postgresql

after that i build from source at 8.2.0.2-466. test with default success.
will test again with 8.2.0.2-466 and postgresql

my error 
`2019-01-21 21:30:46,551 ERROR [org.pentaho.platform.util.logging.Logger] misc-org.pentaho.platform.engine.core.system.PentahoSystem: org.pentaho.platform.api.engine.PentahoSystemException: PentahoSystem.ERROR_0014 - Error while trying to execute startup sequence for org.pentaho.platform.engine.services.connection.datasource.dbcp.DynamicallyPooledDatasourceSystemListener
org.pentaho.platform.api.engine.PentahoSystemException: org.pentaho.platform.api.engine.PentahoSystemException: PentahoSystem.ERROR_0014 - Error while trying to execute startup sequence for org.pentaho.platform.engine.services.connection.datasource.dbcp.DynamicallyPooledDatasourceSystemListener
`
